src/Avalonia.Controls/Primitives/PopupRoot.cs

```diff
@@ -77,7 +77,7 @@ public PopupRoot(TopLevel parent, IPopupImpl impl, IAvaloniaDependencyResolver?
         /// <summary>
         /// Gets the control that is hosting the popup root.
         /// </summary>
-        Visual? IHostedVisualTreeRoot.Host => VisualParent;
+        Visual? IHostedVisualTreeRoot.Host => Parent as Visual ?? ParentTopLevel;
```

Comment:
@Gillibald this fixes what I think was an incorrect change in [one of your commits](https://github.com/AvaloniaUI/Avalonia/commit/6210018f3b5e98063252d2bf2933e1739ed17f85#diff-b9c372514271a3da80207200fb37c0f5e9ef566ec7bff8db21f9190c6a1540cbR80). I think the change in that commit was wrong because `VisualParent` should always be `null` for `PopupRoot` considering that it's a visual root.